ros2run/ros2run/api/__init__.py

```diff
@@ -49,11 +49,21 @@ def get_executable_paths(*, node, package_name):
     return executable_paths
 
 
-def get_executable_path(*, node, package_name, executable_name):
-    paths = get_executable_paths(node=node, package_name=package_name)
-    paths2base = {
-        p: os.path.basename(p) for p in paths
-        if os.path.basename(p) == executable_name}
+def get_executable_path(*, package_name, executable_name):
+    paths = get_executable_paths(package_name=package_name)
+    paths2base = {}
+    for p in paths:
+        basename = os.path.basename(p)
+        if basename == executable_name:
+            # pick exact match
+            paths2base[p] = basename
+        elif sys.platform == 'win32':
```

Comment:
oh right, yeah I saw it the other way around, if I passed `talker.exe` it would pick that rather than trying to remove the extension and look for generic extension points without extension. But it does make a lot more sense as you described it :confounded: